airflow/contrib/operators/azure_hdinsight_create_cluster_operator.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import logging
+
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.resource.resources.models import DeploymentMode, DeploymentProperties
+
+from airflow.models import BaseOperator
+
+
+class HDInsightCreateClusterOperator(BaseOperator):
+    """
+    Refences
+    https://github.com/Azure-Samples/resource-manager-python-resources-and-groups
+    https://docs.microsoft.com/en-us/python/api/overview/azure/key-vault?toc=%2Fpython%2Fazure%2FTOC.json&view=azure-python
+    """
+
+    def __init__(self,
+                 client_id,
+                 secret,
+                 tenant,
+                 subscription_id,
+                 resource_group_name,
+                 resource_group_location,
+                 deploy_name,
+                 template_json,
+                 parameters_json,
+                 *args,
+                 **kwargs
+                 ):
+        super(HDInsightCreateClusterOperator, self).__init__(*args, **kwargs)
+        self.secret = secret
+        self.client_id = client_id
+        self.tenant = tenant
+        self.subscription_id = subscription_id
+        self.resource_group_name = resource_group_name
+        self.resource_group_location = resource_group_location
+        self.deploy_name = deploy_name
+        self.template_json = template_json
+        self.parameters_json = parameters_json
+
+    def create_cluster(self, ):
+        client = ResourceManagementClient(self.get_credential(), self.subscription_id)
```

Comment:
Operator should not create the SDK ibrary directly. It should invoke the method from the hook that deals with the management of the connection. This solution will allow you to perform API operation in other operators.